addons/mail/static/src/model/model_manager.js

```diff
@@ -639,6 +517,186 @@ class ModelManager {
         }
     }
 
+    /**
+     * @private
+     * @param {mail.model} Model class
+     * @param {Object|Object[]} [data={}]
+     * @returns {mail.model|mail.model[]}
+     */
+    _create(Model, data = {}) {
+        const isMulti = typeof data[Symbol.iterator] === 'function';
+        const dataList = isMulti ? data : [data];
+        const fieldNames = new Set(Object.keys(Model.fields));
+        const fields = Object.values(Model.fields);
+        const records = [];
+        for (const data of dataList) {
+            // Make proxified record, so that access to field redirects
+            // to field getter.
+            const record = new Proxy(new Model({ valid: true }), {
+                get: (target, k) => {
+                    if (!(fieldNames.has(k))) {
+                        // No crash, we allow these reads due to patch()
+                        // implementation details that read on `this._super` even
+                        // if not set before-hand.
+                        return target[k];
+                    }
+                    return Model.fields[k].get(target);
+                },
+                set: (target, k, newVal) => {
+                    if (fieldNames.has(k)) {
+                        throw new Error("Forbidden to write on record field without .update()!!");
+                    } else {
+                        // No crash, we allow these writes due to following concerns:
+                        // - patch() implementation details that write on `this._super`
+                        // - record listeners that need setting on this with `.bind(this)`
+                        target[k] = newVal;
+                    }
+                    return true;
+                },
+            });
+            record.env = this.env;
+            record.localId = Model._createRecordLocalId(data);
+            if (Model.get(record.localId)) {
+                throw Error(`A record already exists for model "${Model.modelName}" with localId "${record.localId}".`);
+            }
+            // Contains field values of record.
+            record.__values = {};
+            // Contains revNumber of record for checking record update in useStore.
+            record.__state = 0;
+
+            Model.__records[record.localId] = record;
+
+            // Ensure X2many relations are Set initially (other fields can stay undefined).
+            for (const field of fields) {
+                if (field.fieldType === 'relation') {
+                    if (['one2many', 'many2many'].includes(field.relationType)) {
+                        record.__values[field.fieldName] = new Set();
+                    }
+                }
+            }
+
+            record._willCreate(data);
+
```

Comment:
no empty lines around, could have comment above if you feel so (like `// invoke '_willCreate' hook`)